srcCppCommon/PerftestPrinter.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * (c) 2005-2020  Copyright, Real-Time Innovations, Inc. All rights reserved.
+ * Subject to Eclipse Public License v1.0; see LICENSE.md for details.
+ */
+
+#ifndef __PERFTESTPRINTER_H__
+#define __PERFTESTPRINTER_H__
+
+#ifdef RTI_LANGUAGE_CPP_TRADITIONAL
+  #include "Infrastructure_common.h"
+#elif defined(RTI_LANGUAGE_CPP_MODERN)
+  #define PERFT_TIME_UNIT "us"
+#endif
+#include <iostream>
+#include <stdio.h>
+#include <string>
+
+enum PerftestOuputFormat { LEGACY, JSON, CSV };
+
+class PerftestPrinter {
+private:
+    unsigned int _dataLength;
+    bool _printHeaders;
+    bool _showCPU;
+    bool _printIntervals;
+    bool _printSummaryHeaders;
+    bool _printSerialization;
+    bool _isJsonInitialized;
+    bool _controlJsonIntervals;
+    PerftestOuputFormat _outputFormat;
+
+public:
+    PerftestPrinter();
+    ~PerftestPrinter()
+    {
+    }
+
+    void initialize(
+            bool printIntervals,
+            std::string outputFormat,
+            bool printHeaders,
+            bool printSerialization,
+            bool ShowCpu);
+
+    // Set and Get members
+    void set_data_length(unsigned int dataLength);
+    void set_header_printed(bool printHeaders);
+    // Methods
+    void print_latency_header();
+    void print_latency_interval(
+            unsigned long latency,
+            double latencyAve,
+            double latencyStd,
+            unsigned long latencyMin,
+            unsigned long latencyMax,
+            double outputCpu);
+    void print_latency_summary(
+            int totalSampleSize,
+            double latencyAve,
+            double latencyStd,
+            unsigned long latencyMin,
+            unsigned long latencyMax,
+            unsigned long *latencyHistory,
+            unsigned long long count,
+            double serializeTime,
+            double deserializeTime,
+            double outputCpu);
+    void print_latency_summary(
+            double latencyAve,
+            double latencyStd,
+            unsigned long latencyMin,
+            unsigned long latencyMax,
+            unsigned long *latencyHistory,
+            unsigned long long count,
+            double outputCpu);
+    void print_throughput_header();
+    void print_throughput_interval(
+            unsigned long long lastMsgs,
+            unsigned long long mps,
+            double mpsAve,
+            unsigned long long bps,
+            double bpsAve,
+            unsigned long long missingPackets,
+            float missingPacketsPercent,
+            double outputCpu);
+
```

Comment:
For consistency, can we add a new line after every method, as you did in this case.

Reply:
ok